### PR TITLE
Sync spawning commands

### PR DESCRIPTION
**Pull request by tornac1234**
- [x] Fix "spawn" command (broken since Living Large)
- [x] [new] Sync "sub" command, we can now spawn cyclops with a command
~~- [ ] Maybe:  Move those to command callbacks ("/spawn [VEHICLE TYPE] [OPTIONAL COORDINATES]") with OP requirement~~
- [x] Require the moderator perms at least to use the those commands (only local check for now, but should be switched to the future command system once it's ready)

Adds the translation key: "Nitrox_MissingPermission" to be set like so
"You miss the permissions to execute this command [{PERMISSION}]"

### CODE REVIEW

File: NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs [hunk 1]
```diff
@@ -1,19 +1,41 @@
...
+                                                new CodeMatch(OpCodes.Ldloc_0),
+                                                new CodeMatch(OpCodes.Ldarg_0)
+                                            ])
+                                            .SetOpcodeAndAdvance(OpCodes.Ldnull)
+                                            .SetOperandAndAdvance(Reflect.Method(() => WrappedCallback(default)))
+                                            .InstructionEnumeration();
+    }
+
+    public static void WrappedCallback(GameObject prefab)
+    {
```
Comment (dartasen) [inline NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs:35]: What will happen if we spawn an aurora with the sub command ?

File: NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
+                                                new CodeMatch(OpCodes.Ldsfld),
+                                                new CodeMatch(OpCodes.Ldloc_0),
+                                                new CodeMatch(OpCodes.Ldarg_0)
+                                            ])
+                                            .SetOpcodeAndAdvance(OpCodes.Ldnull)
+                                            .SetOperandAndAdvance(Reflect.Method(() => WrappedCallback(default)))
+                                            .InstructionEnumeration();
+    }
+
+    public static void WrappedCallback(GameObject prefab)
```
Comment (dartasen) [inline NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs:34]: Method is named `WrappedCallback `in the patch and `Callback ` in SpawnConsole patch

File: NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs (hunk 1 above)
Comment (tornac1234) [reply NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs:35]: I tried and it just disappeared after some seconds for some reason (same for escapepod), but I'll probably restrict this to spawn cyclops

File: NitroxClient/GameLogic/Vehicles.cs [hunk 3]
```diff
@@ -232,4 +234,11 @@ public static void RemoveNitroxEntitiesTagging(GameObject constructedObject)
             UnityEngine.Object.DestroyImmediate(nitroxEntity);
         }
     }
+
+    public static VehicleWorldEntity MakeVehicleEntity(GameObject constructedObject, NitroxId constructedObjectId, TechType techType, NitroxId constructorId = null)
```
Comment (dartasen) [inline NitroxClient/GameLogic/Vehicles.cs:238]:
```suggestion
    public static VehicleWorldEntity BuildVehicleWorldEntity(GameObject constructedObject, NitroxId constructedObjectId, TechType techType, NitroxId constructorId = null)
```
We usually use Build prefix for method that acts like a """factory"""

**Review by dartasen — APPROVED** [review]

File: NitroxClient/GameLogic/NitroxConsole.cs [hunk 4]
```diff
@@ -48,32 +45,39 @@ public void Spawn(GameObject gameObject)
         }
 
         /// <summary>
-        /// Spawns a Seamoth or an Exosuit
+        /// Spawns Seamoth, Exosuit and Cyclops
```
Comment (Jannify) [inline NitroxClient/GameLogic/NitroxConsole.cs:48]: Minor:
```suggestion
        /// Spawns Seamoth, Exosuit or Cyclops
```

File: NitroxClient/GameLogic/NitroxConsole.cs [hunk 5]
```diff
@@ -48,32 +45,39 @@ public void Spawn(GameObject gameObject)
...
+            Log.Debug($"Spawning vehicle {techType} with id {id} at {gameObject.transform.position}");
         }
 
-        /// <summary>
-        /// Spawns a Pickupable item
-        /// </summary>
-        private void SpawnItem(GameObject gameObject)
+        private void DefaultSpawn(GameObject gameObject)
+        {
+            item.Dropped(gameObject);
```
Comment (Jannify) [inline NitroxClient/GameLogic/NitroxConsole.cs:63]: Can we rename `item` to `items`? It's not really part of this pr but would be better for readability nonetheless.

File: NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs [hunk 6]
```diff
@@ -1,19 +1,63 @@
...
+     */
+    public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> instructions)
+    {
+        return new CodeMatcher(instructions).MatchEndForward([
+                                                new CodeMatch(OpCodes.Stfld),
+                                                new CodeMatch(OpCodes.Ldsfld),
+                                                new CodeMatch(OpCodes.Ldloc_0),
+                                                new CodeMatch(OpCodes.Ldarg_0)
+                                            ])
+                                            .SetOpcodeAndAdvance(OpCodes.Ldnull)
```
Comment (Jannify) [inline NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs:51]: I am guessing that you replace the `Ldarg_0` here to null as it's no longer a local method being called. Would a `nop` also work? I am guessing that `nop` is more clear and can't produce any side effects.

Comment (tornac1234) [reply NitroxPatcher/Patches/Dynamic/SubConsoleCommand_OnConsoleCommand_sub_Patch.cs:51]: Results of the discussion in the Discord:
"newobj constructor takes an object and a native int, the object being the instance (null in our case for  static) and the native int being the pointer given by ldftn.
ldftn will in either way (static or instance method) just give a pointer without the instance mentionned. It's about the instance we give to the constructor"

**Review by Jannify — APPROVED** [review]
LGTM CW
